### PR TITLE
add /api/topics end point with tests and error handling

### PR DESCRIPTION
**Pull request by 2202Hannah**
I have created the /api/topics end point which responds with an array of topic objects. I have used TDD. I have added error handling for non existent end points.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        });
+      });
+  });
+  test("404: responds with an error when passed a non existant end point", () => {
+    return request(app)
+      .get("/api/non-existant")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Route not found");
+      });
```
Comment (corkesw) [inline __tests__/app.test.js]: This test is not specific to this endpoint and should be placed in to it's own describe block.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
+const app = express();
+const { getTopics } = require(`./controllers/topics.controller.js`);
+app.use(express.json());
+
```
Comment (corkesw) [inline app.js:4]: I am aware that you will need to use `express.json()` later on but this is not necessary for this first endpoint and should not be part of our codebase at this point.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  test("return status 200 when successful", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200);
+  });
+  test("return an object with the expected values", () => {
+    return request(app)
+      .get("/api/topics")
+      .then(({ body: topics }) => {
+        expect(topics).toHaveLength(3);
```
Comment (corkesw) [inline __tests__/app.test.js]: I would expect the returned array to be in an object and on a key of topics. You should change the assertion in your test to reflect this.

File: controllers/topics.controller.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const { selectTopics } = require(`../models/topics.model`);
+
+exports.getTopics = (request, response) => {
+  selectTopics().then(topics => {
+    response.status(200).send(topics);
+  });
```
Comment (corkesw) [inline controllers/topics.controller.js:6]: You should always return data in an object with an appropriate key, in this case a key of topics.

**Review by corkesw — COMMENTED** [review]
Hi Hannah, really nice and clear code, just a few changes to make and then you are good to merge.

File: __tests__/app.test.js (hunk 1 above)
Comment (2202Hannah) [reply __tests__/app.test.js]: Hi @corkesw thank you for your feedback. I have made the changes now and pushed them to this branch. Can I go ahead and merge the pull request? Or should I put another /nchelp in to get the changes checked over? Many thanks,
Hannah.

**Pull request merged**